File: src/main/java/quebec/virtualite/backend/services/domain/database/WheelRepository.java
package quebec.virtualite.backend.services.domain.database;

import org.springframework.data.repository.CrudRepository;
import quebec.virtualite.backend.services.domain.entities.WheelEntity;

import java.util.List;
import java.util.Optional;

public interface WheelRepository extends CrudRepository<WheelEntity, Long>
{
    List<WheelEntity> findAll();

    Optional<WheelEntity> findByName(String name);
}

File: src/main/java/quebec/virtualite/backend/services/domain/database/WheelRepository.kt
package quebec.virtualite.backend.services.domain.database

import org.springframework.data.repository.CrudRepository
import quebec.virtualite.backend.services.domain.entities.WheelEntity

interface WheelRepository : CrudRepository<WheelEntity, Long>
{
    fun findByName(name: String): WheelEntity?
}
